File: Visualize_Table/Script2.py
import pandas as pd
import plotly.graph_objects as go

# Load the CSV
data = pd.read_csv('Student_Performance.csv')

# Convert 'Date' column to datetime format
data['Date'] = pd.to_datetime(data['Date'])

# Aggregate Sleep_Hours by Month
data['Month'] = data['Date'].dt.to_period('M')
data2 = data.groupby('Month')['Sleep_Hours'].mean().reset_index()

# Convert 'Month' to string
data2['Month'] = data2['Month'].astype(str)

# Create a Plotly table
fig = go.Figure(data=[go.Table(
    header=dict(values=["Month", "Average Sleep Hours"],
                fill_color='lightblue',
                align='center'),
    cells=dict(values=[data2['Month'], data2['Sleep_Hours']],
               fill_color='lavender',
               align='center'))
])

# Update table layout
fig.update_layout(
    title="Average Sleep Hours by Month",
    title_font=dict(size=20),
    title_x=0.5,
    title_y=0.95,
    title_xanchor='center',
    title_yanchor='top'
)

fig.show()

# Save the figure as PNG
fig.write_image("table_plot.png")

File: Visualize_Table/Script2.R
# Load necessary libraries
library(tidyverse)

# Load the CSV file
data <- read.csv("Student_Performance.csv")

# Convert 'Date' column to Date format
data$Date <- as.Date(data$Date, format = "%m/%d/%Y")

# Create a new variable 'Month' to represent the month
data <- data %>%
  mutate(Month = format(Date, "%Y %b"))

# Using tidyverse, compute the monthly average Sleep_Hours
monthly_avg_sleep_hours <- data %>%
  group_by(Month) %>%
  summarise(Avg_Sleep_Hours = mean(Sleep_Hours, na.rm = TRUE))

# Print the result
print(monthly_avg_sleep_hours)

# Load necessary libraries
library(gt)

# Generate a nice table using gt
monthly_avg_sleep_hours %>%
  gt() %>%
  tab_header(
    title = "Monthly Average Sleep Hours",
    subtitle = "Computed from the provided data"
  ) %>%
  fmt_number(columns = everything(), decimals = 1) 

# Load necessary libraries
library(flextable)

# Generate a nice table using flextable
flextable(monthly_avg_sleep_hours) %>%
  set_caption("Monthly Average Sleep Hours") %>%
  align(align = "center", part = "all")

# Load necessary libraries
library(kableExtra)

# Generate a nice table using kableExtra
monthly_avg_sleep_hours %>%
  kbl() %>%
  kable_paper("striped", full_width = FALSE) %>%
  add_header_above(c("Monthly Average Sleep Hours" = 2), align = "center") %>%
  column_spec(1:2)



# Load necessary libraries
library(flextable)

# Generate a nice table using flextable
ft <- flextable(monthly_avg_sleep_hours) %>%
  set_caption("Monthly Average Sleep Hours") %>%
  align(align = "center", part = "all") %>%
  theme_booktabs() %>%
  theme_vanilla() %>%
  fontsize(size = 10) %>%
  padding(padding = 2) %>%
  bold(part = "header") %>%
  autofit() 
ft
# Save the flextable as an image
save_as_image(ft, "nice_table.png")











